main.tf

```diff
@@ -65,104 +84,31 @@ resource "google_compute_instance" "elasticsearch" {
 
   network_interface {
     network = var.network
-
-    access_config {
-      // Ephemeral IP
-    }
   }
 
   metadata = {
-    ssh-keys = "devops:${tls_private_key.provision.public_key_openssh}"
+    ssh-keys  = "devops:${tls_private_key.provision.public_key_openssh}"
+    user-data = <<-EOT
+#!/bin/bash
+
+echo "${base64encode(local.elasticsearch_configuration)}" | base64 -d > /tmp/elasticsearch.yml;
+echo "${google_service_account_key.elasticsearch_backup.private_key}" | base64 -d > /tmp/backup-sa.key;
+echo "${base64encode(local.master_list)}" | base64 -d > /home/devops/master_list.sh;
```

Comment:
1. Why can't we just:

```bash
 echo "${local.master_list}" > /home/devops/master_list.sh;
```

2. Does the `.sh` extension make sense?

Reply:
It's called as `source /home/devops/master_list.sh` and later used in here:
```
-Ecluster.initial_master_nodes=${master_list}
```
we can probably export the variable straight forward in `user-data` and not use the file at all.